### PR TITLE
Move files from root folder to config folder

### LINKED ISSUE #1248: Root directory cleanup
**Is your feature request related to a problem? Please describe.**

We have too many extraneous executable files in the root directory of the repo. This causes some risk:

1. Increases the likelihood of someone deleting a file or executing it accidentally or incorrectly
2. Increases the likelihood of files not being used at all because they addressed some legacy requirement

**Describe the solution you'd like**

1. Create a `tools/` directory
1. Migrate all `.ts`, `.tsx`, `.js` and `.py `files to this directory if possible
1. Update the documentation and scripts that rely on these files so that repo works correctly

**Describe alternatives you've considered**
None

**Approach to be followed (optional)**

N/A

**Additional context**

N/A

### PR DESCRIPTION
**Pull request by anshgoyalevil**
**What kind of change does this PR introduce?**
This PR moves the file ```setupFile.ts``` from root folder to config folder.

**Issue Number:**

Fixes #1248 

**Did you add tests for your changes?**
Not Required

**Snapshots/Videos:**
NA

**If relevant, did you update the documentation?**
Not Req.

**Summary**
This PR moves the file ```setupFile.ts``` from root folder to config folder.

**Does this PR introduce a breaking change?**
No

**Other information**
There are several other files in the root of the project. Let me explain why they need to be there only.
```codegen.ts``` - it needs to be in the directory where ```npm run dev``` is executed
```.*``` - these are the configuration files such as ```.eslintrc.json``` which control the project's source code for formatting, linting, etc. They needs to be in the root of the project as a standard coding convention. The node_module for these packages search for these files inside the root of the project.
```installer.py``` and ```requirements.py``` - these files need to be in root of the project since they would help develop configure the project by finding these files in root dir.

Let me know, if you think any of these files still need not be in root dir. 

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**
Yes

### CODE REVIEW

**Comment by EshaanAgg** [comment]
@anshgoyalevil You can rename `setupFile` to `VitestSetup` as the same is more descriptive. The current naming is too vague. 
Also I feel that `config` or any other name name along the same files would be a more descriptive name for this directory instead of `tools`.

**Comment by EshaanAgg** [comment]
Additonally, you can also consider creating an `assets` directory and move the `image` and `locales` folders into the same as they are static assets that wouldn't be changed. 
Even `installer.py` can be moved to be moved to a non root directory. 

What do u think?

**Comment by anshgoyalevil** [comment]
> @anshgoyalevil You can rename `setupFile` to `VitestSetup` as the same is more descriptive. The current naming is too vague. Also I feel that `config` or any other name name along the same files would be a more descriptive name for this directory instead of `tools`.

Hmm. The ```config``` would look good. Will also change the name of setupFile to VitestSetup

**Comment by anshgoyalevil** [comment]
> Additonally, you can also consider creating an `assets` directory and move the `image` and `locales` folders into the same as they are static assets that wouldn't be changed. Even `installer.py` can be moved to be moved to a non root directory.
> 
> What do u think?

I think ```image``` and ```locales``` can be moved to ```assets``` folder. 

For ```installer.py```, it shouldn't be moved since it is the first file a python developer would search for after cloning the repo.

**Comment by Cioppolo14** [comment]
@noman2002 @xoldyckk I added reviewers even though the unauthorized file check failed because I think it is a required change for this issue.

**Review by xoldyckk — APPROVED** [review]

**Review by palisadoes — APPROVED** [review]

File: vite.config.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { defineConfig } from "vitest/config";
 
 export default defineConfig({
   test: {
-    setupFiles: ["./setupFile.ts"],
+    setupFiles: ["./config/vitestSetup.ts"],
```
[the remark this answers is not in the document]
Comment (anshgoyalevil) [reply vite.config.ts:5]: I think, its correct. This file is used to configure that tests fail loudly. So, placing it inside ```config``` folder, and naming it ```vitestSetup.ts``` is justified.

This file extends the vitest functionality, and setup the tests before running the ```npm run test``` command.

**Pull request merged**